shuttle_test.go

```diff
@@ -34,23 +23,88 @@ Felis ligula, malesuada suscipit malesuada non, ultrices non.
 Shomer shabbos.
 Urna sed orci ipsum, placerat id condimentum rutrum, rhoncus.
 Yeah man, it really tied the room together.
-Ac lorem aliquam placerat.`))}
+Ac lorem aliquam placerat.
+`)
+
+func newTestConfig() Config {
+	// Defaults should be good for most tests
+	config := NewConfig()
+	config.LogsURL = "http://"
+	return config
 }
 
-func NewTestInput() *TestInput {
-	return &TestInput{bytes.NewReader([]byte(`Hello World
-Test Line 2
-`))}
+type loopingBuffer struct {
+	b     []byte
+	close chan struct{}
+	p     int
+
+	mu *sync.Mutex
```

Comment:
Similar to before, why a pointer? If it was just a value it would be ready to use without any initialization.